Sources/XiEditor/EditView.swift

```diff
@@ -604,6 +613,22 @@ final class EditView: NSView, NSTextInputClient, TextPlaneDelegate {
 
                 let x = dataSource.gutterWidth - (gutterXPad + CGFloat(gutterTL.width))
                 let y0 = yOff + dataSource.textMetrics.ascent + linespace * CGFloat(lineIx)
+
+                for annotationType in gutterAnnotations {
+                    var annotations = annotationsForLines[lineIx]![annotationType]!
+                    let annotation = annotations.next()
+
+                    if annotation != nil {
```

Comment:
In Swift there's a `while let` syntax that's kind of similar to Rust's, so this could be written as
`while let annotation = annotations.next()`.

I also think repeating `annotation` is a bit confusing. Maybe this could be `lineAnnotation`?